plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/settings/CodeWhispererConfigurable.kt

```diff
@@ -87,6 +89,55 @@ class CodeWhispererConfigurable(private val project: Project) :
             }
         }
 
+        group(message("aws.settings.codewhisperer.group.q_chat")) {
+            row {
+                checkBox(message("aws.settings.codewhisperer.project_context")).apply {
+                    connect.subscribe(
+                        ToolkitConnectionManagerListener.TOPIC,
+                        object : ToolkitConnectionManagerListener {
+                            override fun activeConnectionChanged(newConnection: ToolkitConnection?) {
+                                enabled(isCodeWhispererEnabled(project))
+                            }
+                        }
+                    )
+                    enabled(invoke)
+                    bindSelected(codeWhispererSettings::isProjectContextEnabled, codeWhispererSettings::toggleProjectContextEnabled)
+                }.comment(message("aws.settings.codewhisperer.project_context.tooltip"))
+            }
+
+            row {
+                intTextField(
+                    range = IntRange(0, 1000)
+                ).bindIntText(codeWhispererSettings::getProjectContextIndexThreadCount, codeWhispererSettings::setProjectContextIndexThreadCount)
+                    .align(AlignX.FILL).apply {
+                        connect.subscribe(
+                            ToolkitConnectionManagerListener.TOPIC,
+                            object : ToolkitConnectionManagerListener {
+                                override fun activeConnectionChanged(newConnection: ToolkitConnection?) {
+                                    enabled(isCodeWhispererEnabled(project))
+                                }
+                            }
+                        )
+                        enabled(invoke)
```

Comment:
should the thread setting be available if the user is using GPU?

Reply:
this thread setting is only for CPU, we will consider adding another line for that in description

Reply:
if it's either CPU or GPU and not at the same time, then i think it should be a radio button rather than a checkbox

Reply:
>if it's either CPU or GPU and not at the same time

It is in fact not so straightforward. For each different GPU, it may support different NN operators. The model, after being transformed into a NN operator graph, are distributed to both CPU and GPU for computation. For the majority of NN operators, if they can be run on GPU, they will. 

I notice that at least in Windows with AMD GPU, 100% of NN operator is moved to GPU. Not sure about intel GPU. It should be 100% for NVIDIA GPU. But for Mac OS GPU, it can only load 10% of the NN operators, which resulted in CPU running at 90% and GPU at 16%.